File: Java/6kyu/take-a-ten-minute-walk.java
// Kata link: https://www.codewars.com/kata/take-a-ten-minute-walk


public class TenMinWalk {

    public static boolean isValid(char[] walk) {
        int dx = 0;
        int dy = 0;

        if (walk.length == 10) {
            for (int i = 0; i < walk.length; i++)
                switch (walk[i]) {
                    case 'n': dy++; break;
                    case 's': dy--; break;
                    case 'w': dx--; break;
                    case 'e': dx++; break;
                }
            return dx == 0 && dy == 0;
        }

        return false;
    }

}

File: Javascript/6kyu/take-a-ten-minute-walk.js
// Kata link: https://www.codewars.com/kata/take-a-ten-minute-walk


function isValidWalk(walk) {
    return walk.length === 10 &&
        walk.reduce((route, v, i) => {
            if (v === 'n')
                ++route.y;
            else if (v === 's')
                --route.y;
            else if (v === 'e')
                ++route.x;
            else if (v === 'w')
                --route.x;

            return i === walk.length - 1
                ? (route.x === 0) && (route.y === 0)
                    ? true
                    : false
                : route;
        },
        {
            x: 0,
            y: 0
        });
}
